pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -309,6 +317,9 @@ func (tm *TableManager) ensureQueryReadiness(ctx context.Context) error {
 			return err
 		}
 
+		usersToBeQueryReadyLen += len(usersToBeQueryReadyFor)
+
+		level.Debug(util_log.Logger).Log("msg", "instance should be query ready for users", "users", usersToBeQueryReadyFor)
```

Comment:
we could also log `time.Since(start)` here?

Reply:
good call. I'll add it.